src/nodes/android/armodel.ts

```diff
@@ -19,8 +32,25 @@ export class ARModel extends ARCommonGeometryNode {
             }
           }))
           .exceptionally(new java.util.function.Function({
-              apply: error => reject(error)
+
+              apply: error => {
+                 console.error("failed loading: "+options.name);
+                reject(error);
+              }
           }));
     });
   }
+
+
+
+  static getGLTFSource(asset:string): any {
+    const context = utils.ad.getApplicationContext();
+
+
+        const type=(asset.indexOf(".glb")>0)?(<any>com.google.ar.sceneform).assets.RenderableSource.SourceType.GLB:(<any>com.google.ar.sceneform).assets.RenderableSource.SourceType.GLTF;
```

Comment:
@nickolanack According to https://developers.google.com/ar/reference/java/sceneform/reference/com/google/ar/sceneform/assets/RenderableSource.SourceType?hl=nl the `GLTF` enum is actually `GLTF2`. Would that perhaps fix the crash?

Reply:
On a related note: I'll try to generate TypeScript typings for those classes, so we can catch this kind of issues at code-time.

Reply:
Oh wow. Nice catch! I’ll give that a shot

Reply:
yup that fixes it!